RealmSwift/List.swift

```diff
@@ -118,28 +118,28 @@ public final class List<T: Object>: ListBase {
     public subscript(position: Int) -> T {
         get {
             throwForNegativeIndex(position)
-            return unsafeBitCast(_rlmArray.object(at: UInt(position)), to: T.self)
+            return cast(_rlmArray.object(at: UInt(position)), to: T.self)
         }
         set {
             throwForNegativeIndex(position)
-            _rlmArray.replaceObject(at: UInt(position), with: newValue.unsafeCastToRLMObject())
+            _rlmArray.replaceObject(at: UInt(position), with: newValue as AnyObject)
         }
     }
 
     /// Returns the first object in the list, or `nil` if the list is empty.
-    public var first: T? { return unsafeBitCast(_rlmArray.firstObject(), to: Optional<T>.self) }
+    public var first: T? { return cast(_rlmArray.firstObject(), to: Optional<T>.self) }
 
     /// Returns the last object in the list, or `nil` if the list is empty.
-    public var last: T? { return unsafeBitCast(_rlmArray.lastObject(), to: Optional<T>.self) }
+    public var last: T? { return cast(_rlmArray.lastObject(), to: Optional<T>.self) }
 
     // MARK: KVC
 
     /**
      Returns an `Array` containing the results of invoking `valueForKey(_:)` using `key` on each of the collection's
      objects.
-    */
-    public override func value(forKey key: String) -> Any? {
-        return value(forKeyPath: key)
+     */
+    @nonobjc public func value(forKey key: String) -> [AnyObject] {
```

Comment:
Does `@nonobjc` mean that this cannot be invoked from Obj-C? I'd think that would give unexpected behavior from calls to `-valueForKeyPath:` with a key path that traverses into a `List` but where the `List` isn't the root object.